server/lib/src/idm/server.rs

```diff
@@ -4125,4 +4126,92 @@ mod tests {
 
         // Any checks?
     }
+
+    async fn idm_fallback_auth_fixture(
+        idms: &IdmServer,
+        _idms_delayed: &mut IdmServerDelayed,
+        has_posix_password: bool,
+        allow_primary_cred_fallback: Option<bool>,
+        expected: Option<()>,
+    ) {
+        let ct = Duration::from_secs(TEST_CURRENT_TIME);
+        let target_uuid = Uuid::new_v4();
+        let p = CryptoPolicy::minimum();
+
+        {
+            let mut idms_prox_write = idms.proxy_write(ct).await.unwrap();
+
+            let mut e = entry_init!(
+                (Attribute::Class, EntryClass::Object.to_value()),
+                (Attribute::Class, EntryClass::Account.to_value()),
+                (Attribute::Class, EntryClass::Person.to_value()),
+                (Attribute::Name, Value::new_iname("kevin")),
+                (Attribute::DisplayName, Value::new_utf8s("Kevin")),
+                (Attribute::Uuid, Value::Uuid(target_uuid)),
+                (
+                    Attribute::PrimaryCredential,
+                    Value::Cred(
+                        "primary".to_string(),
+                        Credential::new_password_only(&p, "banana").unwrap()
+                    )
+                )
+            );
+
+            if let Some(allow_primary_cred_fallback) = allow_primary_cred_fallback {
```

Comment:
AllowPrimaryCredFallback is part of account policy - account policy goes on groups, not users. So you need to create a group, or attach this to IDM_ALL_ACCOUNTS which is an easy way to do this in tests https://github.com/kanidm/kanidm/blob/983135e353aa8e69385fa45fef5082992221f328/server/lib/src/idm/credupdatesession.rs#L3697